lib/fetch/dataURL.js

```diff
@@ -117,25 +117,37 @@ function URLSerializer (url, excludeFragment = false) {
 
 // https://infra.spec.whatwg.org/#collect-a-sequence-of-code-points
 /**
- * @param {(char: string) => boolean} condition
- * @param {string} input
+ * @template {string|Buffer} T
+ * @param {(char: T[number]) => boolean} condition
+ * @param {T} input
  * @param {{ position: number }} position
+ * @returns {T}
  */
 function collectASequenceOfCodePoints (condition, input, position) {
+  const inputIsString = typeof input === 'string'
+  const start = position.position
+
   // 1. Let result be the empty string.
   let result = ''
 
   // 2. While position doesn’t point past the end of input and the
   // code point at position within input meets the condition condition:
   while (position.position < input.length && condition(input[position.position])) {
     // 1. Append that code point to the end of result.
-    result += input[position.position]
+    if (inputIsString) {
+      result += input[position.position]
```

Comment:
If inputIsString you might skip the while loop and provide a fast path for this function.

Reply:
We can't skip the loop entirely, we need to increase the index and check if the character matches the condition